File: js/src/sorting/Insertion.js
import Sorter from './Sorter';

class Insertion extends Sorter {
    constructor(arr = []) {
        super(arr);
    }

    sort() {
        let N = this.arr.length;
        for (let i = 1; i < N; i++) {
            for (let j = i; j >0 &&  this.less(this.arr[j], this.arr[j-1]); j--) {
                this.exch(j, j-1);
            }
        }
    }
}

module.exports = Insertion;

File: java/src/com/test/sorting/Insertion.java
package com.test.sorting;


public class Insertion extends Sorter
{
    public Insertion(int[] arr)
    {
        super(arr);
    }

    @Override
    public void sort()
    {
        int N = arr.length;
        for (int i = 1; i < N; i++) {
            for (int j = i; j > 0 && less(arr[j], arr[j-1]); j--) {
                exch(j, j-1);
            }
        }
    }
}
